lua/tardis/parts/default_controls.lua

```diff
@@ -0,0 +1,704 @@
+
+local PART={}
+PART.ID = "default_gears1"
+PART.Model = "models/molda/toyota_int/gears1.mdl"
+PART.AutoSetup = true
+PART.Collision = true
+PART.ShouldTakeDamage = false
+
+PART.Animate = true
+PART.AnimateOptions = {
+    Type = "perpetual_use",
+    Speed = 1.2,
+    StopAnywhere = true,
+    SpeedOverrideFunc = function(a, target, do_reset)
+        if do_reset and (a.pos < 0.2 or a.pos > 0.8) then
+            return a.speed * 3
+        end
+        return a.speed
+    end,
+}
+
+if CLIENT then
+    PART.Use = function(self, ply)
+        self.animation.stop_anywhere = LocalPlayer():KeyDown(IN_WALK)
+    end
+end
+
+TARDIS:AddPart(PART)
+
+PART.ID = "default_gears2"
+PART.Model = "models/molda/toyota_int/gears2.mdl"
+PART.AnimateOptions.Speed = 1.5
+TARDIS:AddPart(PART)
+
+PART.ID = "default_gears3"
+PART.Model = "models/molda/toyota_int/gears3.mdl"
+PART.AnimateOptions.Speed = 0.9
+TARDIS:AddPart(PART)
+
+local PART={}
+PART.ID = "default_handbrake"
+PART.Model = "models/molda/toyota_int/handbrake.mdl"
+PART.SoundOn = "p00gie/tardis/default/handbrake_on.ogg"
+PART.SoundOff = "p00gie/tardis/default/handbrake_off.ogg"
+PART.AutoSetup = true
+PART.Collision = true
+PART.ShouldTakeDamage = false
+
+PART.EnabledOnStart = true
+
+PART.Animate = true
+PART.AnimateOptions = {
+    Type = "toggle",
+    Speed = 1.2,
+}
+
+if CLIENT then
+
+    function trace_height(int, ply)
+        local trace = {
+            start = ply:EyePos(),
+            endpos = ply:EyePos() + (ply:GetAimVector() * 4096),
+            filter = { ply, },
+        }
+        if util.TraceLine(trace).Hit then
+            return int:WorldToLocal(util.TraceLine(trace).HitPos).z
+        end
+    end
+
+    function PART:Use(ply)
+        if not self:GetOn() then return end
+
+        if LocalPlayer():KeyDown(IN_WALK) then
+            -- giving the player some control over the height
+            local h = trace_height(self.interior, ply) - 132
+            self.animation.max = math.Clamp(0.4 + h * 0.16, 0.4, 1)
+        else
+            self.animation.max = math.Rand(0.5, 1)
+        end
+    end
+end
+
+TARDIS:AddPart(PART)
+
+local PART={}
+PART.ID = "default_throttle"
+PART.Model = "models/molda/toyota_int/throttle.mdl"
+PART.SoundOn = "p00gie/tardis/default/throttle_on.ogg"
+PART.SoundOff = "p00gie/tardis/default/throttle_off.ogg"
+PART.AutoSetup = true
+PART.Collision = true
+PART.ShouldTakeDamage = false
+PART.Animate = true
+PART.AnimateSpeed = 2.3
+TARDIS:AddPart(PART)
+
+local PART={}
+PART.ID = "default_keyboard"
+PART.Model = "models/molda/toyota_int/keyboard.mdl"
+PART.AutoSetup = true
+PART.Collision = true
+PART.ShouldTakeDamage = false
+PART.SoundOn = "p00gie/tardis/default/keyboard_2014.ogg"
+PART.SoundOff = "p00gie/tardis/default/keyboard_2015.ogg"
+PART.SoundNoPower = "p00gie/tardis/default/keyboard.ogg"
+
+if SERVER then
+    function PART:Use(ply)
+        self.interior:Timer("default_keyboard", 1, function()
```

Comment:
I'm not sure about the delays on activating parts with the sounds, it feels sluggish especially on controls like destination select, I thought it was broken cause I had my music on

Reply:
Something we could maybe do to alleviate this a bit is to lock the player view while it's waiting, would make it clear that they have done something then

Reply:
I've done a fade effect for telepathics
I don't see an easy way to do it nicely for the keyboard 
I think it's okay since it's been like this in Toyota for a very long time

Reply:
1b95ab59d65b1740483bcfc1eb8f850603529af2

Reply:
Okay! Yeah the telepathic was the worse, keyboard not so bad